plugins/vpc-tunnel/plugin/commands.go

```diff
@@ -0,0 +1,360 @@
+			err = imds.BlockInstanceMetadataEndpoint()
+			if err != nil {
+				return err
+			}
+		}
+
+		return err
+	})
+
+	if err != nil {
+		log.Errorf("Failed to setup the link: %v.", err)
+		return err
+	}
+
+	// Generate CNI result.
+	// IP addresses, routes and DNS are configured by VPC DHCP servers.
+	result := &cniTypesCurrent.Result{
+		Interfaces: []*cniTypesCurrent.Interface{
+			{
+				Name:    args.IfName,
+				Sandbox: args.Netns,
+			},
+		},
+	}
+
+	log.Infof("Writing CNI result to stdout: %+v", result)
+
+	return cniTypes.PrintResult(result, netConfig.CNIVersion)
+}
+
+// Del is the internal implementation of CNI DEL command.
+// CNI DEL command can be called by the orchestrator agent multiple times for the same interface,
+// and thus must be best-effort and idempotent.
+func (plugin *Plugin) Del(args *cniSkel.CmdArgs) error {
+	// Parse network configuration.
+	netConfig, err := tunnelConfig.New(args)
+	if err != nil {
+		log.Errorf("Failed to parse netconfig from args: %v.", err)
+		return err
+	}
+
+	log.Infof("Executing DEL with netconfig: %+v.", netConfig)
+
+	// Derive names from CNI network config.
+	geneveName := fmt.Sprintf(GeneveLinkNameFormat, netConfig.VNI, netConfig.DestinationPort)
+	tapBridgeName := fmt.Sprintf(bridgeNameFormat, netConfig.DestinationPort)
+	tapLinkName := args.IfName
+
+	// Search for the target network namespace.
+	netns, err := netns.GetNetNS(args.Netns)
+	if err == nil {
+		// In target network namespace...
+		err = netns.Run(func() error {
+			var errs error
+			if netConfig.InterfaceType == tunnelConfig.IfTypeTAP {
+				// Delete the tap link.
+				la := netlink.NewLinkAttrs()
+				la.Name = tapLinkName
+				tapLink := &netlink.Tuntap{LinkAttrs: la}
+				log.Infof("Deleting tap link: %v.", tapLinkName)
+				err = netlink.LinkDel(tapLink)
+				if err != nil {
+					log.Errorf("Failed to delete tap link: %v.", err)
+					errs = multierror.Append(err, errs)
+				}
+			}
+
+			// Delete the geneve link.
+			la := netlink.NewLinkAttrs()
+			la.Name = geneveName
+			geneveLink := &netlink.Geneve{LinkAttrs: la}
+			log.Infof("Deleting geneve link: %v.", geneveName)
+			err = netlink.LinkDel(geneveLink)
+			if err != nil {
+				log.Errorf("Failed to delete geneve link: %v.", err)
+				errs = multierror.Append(err, errs)
+			}
+
+			if netConfig.InterfaceType == tunnelConfig.IfTypeTAP {
+				// Delete the tap bridge.
+				la = netlink.NewLinkAttrs()
+				la.Name = tapBridgeName
+				tapBridge := &netlink.Bridge{LinkAttrs: la}
+				log.Infof("Deleting tap bridge: %v.", tapBridgeName)
+				err = netlink.LinkDel(tapBridge)
+				if err != nil {
+					log.Errorf("Failed to delete tap bridge: %v.", err)
+					errs = multierror.Append(err, errs)
+				}
+			}
+
+			return errs
+		})
+		if err != nil {
+			return err
+		}
+	} else {
+		// Log and ignore the failure. DEL can be called multiple times and thus must be idempotent.
+		log.Errorf("Failed to find netns %s, ignoring: %v.", args.Netns, err)
+	}
```

Comment:
you can reduce the level of indentation here by dealing with this case first. Essentially:

```
netns, err := netns.GetNetNS(args.Netns)
if err != nil {
 log.Errorf("Failed to ...")
 return nil
}

// In target network namespace...
err = netns.Run(func() error {
..
}
}
```